src/renderer/src/method/mod.rs

```diff
@@ -0,0 +1,14 @@
+pub mod forward;
+pub mod deferred;
+
+use gfx;
+
+/// A `Method` is an implemnatnion of a Pass
+pub trait Method<A, T, R, C>
```

Comment:
Shouldn't `Pass` and `Framebuffer` be associated types? This would reduce the mental overhead of reading the trait and improve clarity. Unless there are cases where the same struct implements methods for different passes and framebuffers.
Nitpic: `framebuffer` is a very GL-specific name. I'd prefer `Target`, which happens to match `T`.